docs/connection/connection-spans.md

```diff
@@ -0,0 +1,180 @@
+  * [Relationship with application protocols such as HTTP](#relationship-with-application-protocols-such-as-http)
+  * [Connection retry example](#connection-retry-example)
+
+<!-- tocstop -->
+
+this convention defines two types of spans:
+
+- `connect` span: describes the process of establishing a connection. It corresponds to `connect` function ([Linux or other POSIX systems](https://man7.org/linux/man-pages/man2/connect.2.html) /
+[Windows](https://docs.microsoft.com/windows/win32/api/winsock2/nf-winsock2-connect)).
+- `connection` span: describes the connection lifetime: it starts right after the connection is successfully established and ends when connection terminates.
+
+If `connect` spans ends with an error (connection cannot be established), `connection` span SHOULD NOT be created.
+
+If connection can be reused in multiple independent operations, instrumentation SHOULD create `connection` span as a root span in a new trace. The `connection` span should link to the `connect` span. This allows to avoid associating long-lived connection span with a trace which coincidentally started it.
+
+Both spans SHOULD be of a `CLIENT` kind.
+
+## Span name
+
+The **span names** SHOULD match `connect` or `connection` depending on the span type.
+
+## Attributes
+
+The `connect` and `connection` span share the same list of attributes:
+
+<!-- semconv span_attributes.connection.client(full) -->
+| Attribute  | Type | Description  | Examples  | Requirement Level |
+|---|---|---|---|---|
+| [`error.type`](../attributes-registry/error.md) | string | Describes a class of error the operation ended with. [1] | `econnreset`; `econnrefused`; `address_family_not_supported`; `java.net.SocketException` | Conditionally Required: [2] |
+| [`network.local.port`](../attributes-registry/network.md) | int | Local port number of the network connection. | `65123` | Recommended |
+| [`network.peer.address`](../attributes-registry/network.md) | string | Peer address of the network connection - IP address or Unix domain socket name. | `10.1.2.80`; `/tmp/my.sock` | Required |
+| [`network.peer.port`](../attributes-registry/network.md) | int | Peer port number of the network connection. | `65123` | Conditionally Required: when applicable |
+| [`network.transport`](../attributes-registry/network.md) | string | [OSI transport layer](https://osi-model.com/transport-layer/) or [inter-process communication method](https://wikipedia.org/wiki/Inter-process_communication). [3] | `tcp`; `udp` | Recommended |
+| [`network.type`](../attributes-registry/network.md) | string | [OSI network layer](https://osi-model.com/network-layer/) or non-OSI equivalent. [4] | `ipv4`; `ipv6` | Recommended |
+| [`server.address`](../attributes-registry/server.md) | string | Server domain name if available without reverse DNS lookup; otherwise, IP address or Unix domain socket name. [5] | `example.com`; `10.1.2.80`; `/tmp/my.sock` | Conditionally Required: if available without reverse DNS lookup |
```

Comment:
Is this where we would add `network.protocol.name`, `network.protocol.version`, `tls.version`?

Reply:
Do you think they'd be useful on connection spans/metrics?

`network.protocol.*` describe application-level protocol, not a transport-level thing. 
You can send AMQP or HTTP over the socket connection - the connection does not care and does not need to know. 

For TLS and DNS we'll need a new spans not described in this PR